client/albatross_client.ml

```diff
@@ -457,6 +457,13 @@ let gen_cert (cert, certs, key) key_type name (cmd : Vmm_commands.t) =
       (fun e -> `Msg (Fmt.to_to_string X509.Validation.pp_signature_error e))
       (X509.Signing_request.sign csr ~valid_from ~valid_until ~extensions key issuer)
   in
+  let* () =
+    let encoded = X509.Certificate.encode_der mycert in
+    if String.length encoded > 1 lsl 24 then
+      Error (`Msg "certificate too big for TLS")
```

Comment:
When does this happen? When the unikernel is too big, or if the block data is too big, I guess. Do we have a backup method if this fails? I think I remember talk about streaming the unikernel image / block data in the TLS tunnel. I guess this is not a blocker for this PR though.

Reply:
this happens when the block data/unikernel data is bigger than 2^24, and thus transmitting it over tls leads to truncation.

it is not clear to me whether this is a good spot to put it, esp. with this PR being merged we won't have such big certificates anymore (since all is streamed). I developed this patch for the 2.4.1 release where such a ibg block device would simply fail with a strange error...

Reply:
Thanks, that makes sense. I marked this comment chain as resolved.